build failure on riscv32, undefined reference to `log_test_failed_internal'

### systemd version the issue has been seen with

258.2

### Used distribution

Gentoo

### Linux kernel version used

qemu-user chroot

### CPU architectures issue was seen on

other

### Component

systemd-journald

### Expected behaviour you didn't see

systemd should compile and link

### Unexpected behaviour you saw

link failure

```
FAILED: [code=1] systemd-journald 
riscv32-unknown-linux-gnu-gcc  -o systemd-journald systemd-journald.p/meson-generated_.._src_journal_journald-gperf.c.o systemd-journald.p/src_journal_journald.c.o systemd-journald.p/src_journal_journald-audit.c.o systemd-journald.p/src_journal_journald-client.c.o systemd-journald.p/src_journal_journald-config.c.o systemd-journald.p/src_journal_journald-console.c.o systemd-journald.p/src_journal_journald-context.c.o systemd-journald.p/src_journal_journald-kmsg.c.o systemd-journald.p/src_journal_journald-manager.c.o systemd-journald.p/src_journal_journald-native.c.o systemd-journald.p/src_journal_journald-rate-limit.c.o systemd-journald.p/src_journal_journald-socket.c.o systemd-journald.p/src_journal_journald-stream.c.o systemd-journald.p/src_journal_journald-sync.c.o systemd-journald.p/src_journal_journald-syslog.c.o systemd-journald.p/src_journal_journald-varlink.c.o systemd-journald.p/src_journal_journald-wall.c.o systemd-journald.p/src_journal_fuzz-journald-util.c.o -Wl,--as-needed -Wl,--no-undefined -O2 -pipe -Wl,-O1 -Wl,--as-needed -Wl,-z,pack-relative-relocs -Wl,--no-keep-memory '-Wl,-rpath,$ORIGIN/src/shared' src/shared/libsystemd-shared-258.so -pthread
/usr/lib/gcc/riscv32-unknown-linux-gnu/15/../../../../riscv32-unknown-linux-gnu/bin/ld: warning: -z pack-relative-relocs ignored
/usr/lib/gcc/riscv32-unknown-linux-gnu/15/../../../../riscv32-unknown-linux-gnu/bin/ld: systemd-journald.p/src_journal_fuzz-journald-util.c.o: in function `dummy_manager_new':
fuzz-journald-util.c:(.text+0xd6): undefined reference to `log_test_failed_internal'
/usr/lib/gcc/riscv32-unknown-linux-gnu/15/../../../../riscv32-unknown-linux-gnu/bin/ld: fuzz-journald-util.c:(.text+0x11e): undefined reference to `log_test_failed_internal'
/usr/lib/gcc/riscv32-unknown-linux-gnu/15/../../../../riscv32-unknown-linux-gnu/bin/ld: fuzz-journald-util.c:(.text+0x148): undefined reference to `log_test_failed_internal'
collect2: error: ld returned 1 exit status
```

### Steps to reproduce the problem

Happens reproducibly in the Gentoo installation stage build system

### Additional program output to the terminal or log subsystem illustrating the issue

```sh

```

--- a/src/journal/meson.build
+++ b/src/journal/meson.build
@@ -19,10 +19,15 @@ systemd_journald_extract_sources = files(
         'journald-syslog.c',
         'journald-varlink.c',
         'journald-wall.c',
-        # Build fuzz-journald.c as part of systemd-journald so we only compile it once instead of once per
-        # fuzz test.
-        'fuzz-journald-util.c',
 )
+
+if want_fuzz_tests
+        # Build fuzz-journald-util.c as part of systemd-journald so we only
+        # compile it once instead of once per fuzz test.
+        systemd_journald_extract_sources += files(
+                'fuzz-journald-util.c',
+        )
+endif
 
 journald_gperf_c = custom_target(
         input : 'journald-gperf.gperf',

